pkg/tools/etcd_helper.go

```diff
@@ -354,6 +374,17 @@ func (h *EtcdHelper) AtomicUpdate(key string, ptrToType runtime.Object, ignoreNo
 	}
 }
 
+// Examines the etcd key and prepends the prefix if it isn't already there.
+func (h *EtcdHelper) prefixEtcdKey(key string) string {
```

Comment:
Good question.

What breaks if this just does the join in the case that the PathPrefix is non-empty?

The use case I imagine is to prefix registry entries with something like "/kubernetes.io/clustername", so pods would be under /kubernetes.io/clustername/registry/pods.